File: src/main/java/com/emiyaconsulting/dramapopapi/model/Genre.java
package com.emiyaconsulting.dramapopapi.model;


import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.Objects;

@Entity
@Table(name = "genre")
@ToString
@Getter @Setter @NoArgsConstructor
public class Genre {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "name", nullable = false)
    @NotNull
    private String name;
    @Column(name = "deleted")
    private boolean deleted;
    @Column(name = "date_deleted")
    private Instant dateDeleted;
    @Column(name = "date_added", nullable = false, updatable = false) @CreationTimestamp
    private Instant dateAdded;
    @Column(name = "date_modified") @UpdateTimestamp
    private Instant dateModified;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Genre genre)) return false;

        return id != null && Objects.equals(id, genre.id);
    }

    @Override
    public final int hashCode() {
        return getClass().hashCode();
    }
}

File: src/main/kotlin/com/emiyaconsulting/dramapopapi/entity/Genre.kt
package com.emiyaconsulting.dramapopapi.entity

import jakarta.persistence.*
import org.hibernate.proxy.HibernateProxy

@Entity
@Table(name = "genres")
data class Genre (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val genreID: Long,

    @Column(name = "name", nullable = false)
    val name: String,
) {
    final override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null) return false
        val oEffectiveClass =
            if (other is HibernateProxy) other.hibernateLazyInitializer.persistentClass else other.javaClass
        val thisEffectiveClass =
            if (this is HibernateProxy) this.hibernateLazyInitializer.persistentClass else this.javaClass
        if (thisEffectiveClass != oEffectiveClass) return false
        other as Genre

        return genreID != null && genreID == other.genreID
    }

    final override fun hashCode(): Int =
        if (this is HibernateProxy) this.hibernateLazyInitializer.persistentClass.hashCode() else javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(  name = $name )"
    }
}